docs/examples/minhash-dedupe.md

```diff
@@ -0,0 +1,933 @@
+# MinHash Deduplication of Web Text in Common-Crawl
+
+<a target="_blank" href="https://colab.research.google.com/github/Eventual-Inc/Daft/blob/main/tutorials/minhash_dedupe/minhash_dedupe_common_crawl.ipynb">
+  <img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/>
+</a>
+
+In this notebook we will be performing the MinHash Deduplication algorithm over extracted text from HTML documents in the Common Crawl dataset.
+
+If you google "minhash deduplication" you'll find a variety of sources that can walk you through the algorithm. [Finding Near Duplicates with Jaccard Similarity and MinHash by Nelson Elhage](https://blog.nelhage.com/post/fuzzy-dedup/) is a great place to start, but if you are looking for the canonical reference for the MinHash deduplication algorithm, it originates from the seminal paper by Andrei Z. Broder, published in 1997, titled:
+
+```text
+"On the resemblance and containment of documents"
+Published in: Proceedings of the Compression and Complexity of Sequences 1997 (SEQUENCES '97)
+Publisher: IEEE Computer Society
+DOI: 10.1109/SEQUEN.1997.666900
+```
+
+---
+
+The Common Crawl corpus contains petabytes of data, with its oldest entries dating back to 2008, including raw web page data, metadata extracts, and text extracts.
+
+Deduplication is a helpful top-of-funnel strategy for improving dataset quality and is commonly used to improve generalization in LLM training, RAG, and Search.
+
+---
+
+## Table of Contents
+
+- [Quickstart](#quickstart)
+- [Loading Common Crawl](#loading-html-documents-from-common-crawl)
+- [Preprocessing](#preprocessing)
+- [Text Normalization](#text-normalization)
+- [MinHash](#minhash)
+- [LSH Banding](#lsh-banding)
+- [Connected Components](#connected-components)
+- [Validation with igraph](#validation-with-igraph)
+- [Merge Results](#merge-results)
+- [Conclusion](#conclusion)
+
+## Quickstart
+
+```python
+# Install dependencies (skip if using uv - dependencies are already in pyproject.toml)
+# For Google Colab or standalone environments:
+!pip install 'daft[aws,pandas]' selectolax scipy matplotlib igraph
+```
+
+### Define Key Parameters
+
+```python
+NUM_ROWS = 500
+index_col = "block_id"
+content_col = "block"
+
+# Minhash Parameters
+K = 64 # Number of Permutations
+SEED = 42 # Seed for the hash function
+NGRAM_SIZE = 5 # Size of the n-grams
+LSH_THRESHOLD = 0.7 # Jaccard Similarity Threshold for LSH
+```
+
+---
+
+## Loading HTML Documents from Common Crawl
+
+We will be accessing Common Crawl through [WARC files](https://commoncrawl.org/blog/navigating-the-warc-file-format) since [Daft supports the format natively](https://docs.getdaft.io/en/stable/api/io/#daft.read_warc).
```

Comment:
This creates an external link to our old domain. Would prefer

```suggestion
We will be accessing Common Crawl through [WARC files](https://commoncrawl.org/blog/navigating-the-warc-file-format) since [Daft supports the format natively](../api/io/#daft.read_warc).
```